operators/README.md

```diff
@@ -1,53 +1,91 @@
+Both LabTemplates and LabInstances are **namespaced**. 
 
-If you want to delete the CRD run `make uninstall-lab-template`.
+#### Add CRDs to the cluster
 
-### CRD generation
-To modify the LabInstance CRD you need to
-1. open the file _labInstance-operator/api/v1/labinstance_types.go_. Here you have all the `struct` types related to the LabInstance CRD
-2. add/modify/delete the fields of `LabInstanceSpec` and/or `LabInstanceStatus`
-3. run `make`; this will regenerate the code for the new version of LabInstance
-4. you can find the CRD generate in _config/crd/bases_
+Before the deploying the operator, we have to add the LabInstance and LabTemplate CRDs. This can be done via the Makefile:
+
+```bash
+make install
+```
+
+## Laboratory Instance Operator (LabOperator)
+
+Based on [Kubebuilder 2.3](https://github.com/kubernetes-sigs/kubebuilder.git), the operator implements the laboratory creation logic of Crownlabs.
+
+Upon the creation of a *LabInstance*, the operator triggers the creation of the following components:
+* Kubevirt VirtualMachine Instance and the logic to access the noVNC instance inside the VM (Service, Ingress)
+* An instance of [Oauth2 Proxy](https://github.com/oauth2-proxy/oauth2-proxy) (Deployment, Service, Ingress) to regulate access to the VM.
+
+All those resources are binded to the LabInstance life-cycle via the [OwnerRef property](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)
 
 ### Installation
-1. run `make install`. This will install the CRD LabInstance on your cluster.
-2. in _config/samples_ you have an example of LabInstance object. If you want to create it run `kubectl apply -f config/samples/instance_v1_labinstance.yaml`
-3. you can get the list of CRD installed on your cluster by running `kubectl get crd`.
-4. To get the list of LabInstance resources run `kubectl get labi`.
 
-If you want to delete the CRD run `make uninstall`.
+#### Pre-requirements
+
+The only LabOperator requirement is to have Kubevirt deployed.
+This can be done with the following commands, as reported by the official website:
+
+```bash
+# On other OS you might need to define it like
+export KUBEVIRT_VERSION="v0.34.0"
+
+# Deploy the KubeVirt operator
+kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-operator.yaml
+# Only if HW Virtualization is not available
+kubectl create configmap kubevirt-config -n kubevirt --from-literal debug.useEmulation=true
+# Deploy Kubevirt
+kubectl create -f https://github.com/kubevirt/kubevirt/releases/download/${KUBEVIRT_VERSION}/kubevirt-cr.yaml
+```
+
+#### Deployment
+To deploy the LabOperator in your cluster, you have to do the following steps.
+
+First, set the desired values in `operators/labInstance-operator/k8s-manifest-example.env` .
+
+Then export the environment variables and generate the manifest from the template using:
+```
+export $(xargs < k8s-manifest-example.env)
```

Comment:
I would add `cd operators/deploy/laboratory-operator` at the beginning, just for clarity